src/runner.ts

```diff
@@ -203,34 +232,40 @@ function _return(
         _run,
       )
     }
-  })
+  }
 
-  return this
+  return createRunner(
+    {
+      ...state,
+      assertions: [...state.assertions, newAssertion],
+    },
+    saga,
+    args,
+  )
 }
 
 interface Throwable {
   pattern?: ErrorPattern
 }
```

Comment:
This shape is now useless since we always pass the whole state to allow creating new instance each time.